tests/ws.c

```diff
@@ -29,15 +30,58 @@
 
 #include "testutil.h"
 
+#define SOCKET_ADDRESS  "ws://127.0.0.1:5555"
+
 /*  Basic tests for WebSocket transport. */
 
+/*  test_text() verifies that we drop messages properly when sending invalid
+    UTF-8, but not when we send valid data. */
+void test_text() {
+
+    int sb;
+    int sc;
+    int opt;
+    uint8_t bad[20];
+
+    /*  Negative testing... bad UTF-8 data for text. */
+    sb = test_socket (AF_SP, NN_PAIR);
+    sc = test_socket (AF_SP, NN_PAIR);
+
+    /*  Wait for connects to establish. */
+    nn_sleep (200);
+
+    opt = NN_WS_MSG_TYPE_TEXT;
+    test_setsockopt(sb, NN_WS, NN_WS_MSG_TYPE, &opt, sizeof (opt));
+    opt = NN_WS_MSG_TYPE_TEXT;
+    test_setsockopt(sc, NN_WS, NN_WS_MSG_TYPE, &opt, sizeof (opt));
+    opt = 500;
+    test_setsockopt(sb, NN_SOL_SOCKET, NN_RCVTIMEO, &opt, sizeof (opt));
+
+    test_bind (sb, SOCKET_ADDRESS);
+    test_connect (sc, SOCKET_ADDRESS);
+
+    test_send (sc, "GOOD");
+    test_recv (sb, "GOOD");
+
+    /*  and the bad ... */
+    strcpy((char *)bad, "BAD.");
+    bad[2] = (char)0xDD;
+    test_send (sc, (char *)bad);
+
+    /*  Make sure we dropped the frame.  Apparently it comes back as EAGAIN
+        rather than ETIMEDOUT.  I think this possibly a doc bug. */
```

Comment:
Right.  The intention is that new test “test_drop()” tests that a message
was dropped.  I do intend to use this in future tests. :-)
 recv_fail_with_error suggests that it gets back an error, but in this case
its really “recv_got_nothing”. :-)  Which is mostly why I like
“test_drop()”.  To me at least, this is a concise description of what its
testing for.

On Sun, Oct 25, 2015 at 9:01 AM, Wirebird Labs LLC <notifications@github.com
> wrote:

> In tests/ws.c
> <https://github.com/nanomsg/nanomsg/pull/492#discussion_r42946368>:
>
> > +    opt = 500;
> > +    test_setsockopt(sb, NN_SOL_SOCKET, NN_RCVTIMEO, &opt, sizeof (opt));
> > +
> > +    test_bind (sb, SOCKET_ADDRESS);
> > +    test_connect (sc, SOCKET_ADDRESS);
> > +
> > +    test_send (sc, "GOOD");
> > +    test_recv (sb, "GOOD");
> > +
> > +    /*  and the bad ... */
> > +    strcpy((char *)bad, "BAD.");
> > +    bad[2] = (char)0xDD;
> > +    test_send (sc, (char *)bad);
> > +
> > +    /*  Make sure we dropped the frame.  Apparently it comes back as EAGAIN
> > +        rather than ETIMEDOUT.  I think this possibly a doc bug. */
>
> The recv is timing out due to no input
>
> the recv on which line? I just see a send on line 69.
>
> I now found it within the new test_drop. This is a nit/opinion, but
> perhaps that function could be renamed test_recv_fail_with_error? or
> similar? such function/macro could be useful in making many tests more
> concise.
>
> (my misunderstanding is due to code reviews on github, where it's easier
> to overlook detail, rather than being in an IDE with richer context)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nanomsg/nanomsg/pull/492/files#r42946368>.
>